Java/policy-service/src/main/java/com/insuranceapplication/policyservice/methods/PremiumCalculation.java

```diff
@@ -1,234 +1,209 @@
-                }
-            }
-            if (calcConfig.getCombinationName().equals("license_age")) {
-                if (getPeriod(insuredDriver.getD01()) < Integer.valueOf(calcConfig.getValue1())
-                        && calcConfig.getComboId().equals("LIC_BE")) {
-                    calculatedPremium = calculatedPremium + Integer.valueOf(calcConfig.getValue2());
-                }
-                if (getPeriod(insuredDriver.getD01()) < Integer.valueOf(calcConfig.getValue1())
-                        && getPeriod(insuredDriver.getD01()) >= Integer.valueOf(calcConfig.getValue2())
-                        && calcConfig.getComboId().equals("LIC_LBE")) {
-                    calculatedPremium = calculatedPremium + Integer.valueOf(calcConfig.getValue3());
-                }
-                if (getPeriod(insuredDriver.getD01()) < Integer.valueOf(calcConfig.getValue1())
-                        && calcConfig.getComboId().equals("LIC_L")) {
-                    calculatedPremium = calculatedPremium + Integer.valueOf(calcConfig.getValue2());
-                }
-            }
-
-            if (calcConfig.getCombinationName().equals("car_age")) {
-                if (calcConfig.getComboId().equals("CAR_BE")) {
-                    if (getPeriod(insuredVehicle.getD01()) >= Integer.valueOf(calcConfig.getValue1())) {
-                        calculatedPremium = calculatedPremium + precentToPremium(calcConfig.getValue2(), insuredVehicle.getN02());
-                    }
-                }
-                if (calcConfig.getComboId().equals("CAR_LBE") && calcConfig.getId().equals(Integer.valueOf(30))) {
-                    if (getPeriod(insuredVehicle.getD01()) < Integer.valueOf(calcConfig.getValue1()) && getPeriod(insuredVehicle.getD01()) >= Integer.valueOf(calcConfig.getValue2())) {
-                        calculatedPremium = calculatedPremium + precentToPremium(calcConfig.getValue3(), insuredVehicle.getN02());
-                    }
-                }
-                if (calcConfig.getComboId().equals("CAR_LBE") && calcConfig.getId().equals(Integer.valueOf(29))) {
-                    if (getPeriod(insuredVehicle.getD01()) < Integer.valueOf(calcConfig.getValue1()) && getPeriod(insuredVehicle.getD01()) >= Integer.valueOf(calcConfig.getValue2())) {
-                        calculatedPremium = calculatedPremium + precentToPremium(calcConfig.getValue3(), insuredVehicle.getN02());
-                    }
-                }
-                if (calcConfig.getComboId().equals("CAR_L")) {
-                    if (getPeriod(insuredVehicle.getD01()) < Integer.valueOf(calcConfig.getValue1())) {
-                        calculatedPremium = calculatedPremium + precentToPremium(calcConfig.getValue2(), insuredVehicle.getN02());
-                    }
-                }
-            }
+    public void updatePremium(Double valueOfUpdate, ObjectRisk riskToUpdate) {
```

Comment:
Lepszą nazwa zmiennej określającej nowe premium byłoby newPremium. valueOfUpdate to nazwa która się może tyczyć absolutnie każdej rzeczy w bazie danych.